itests/kit/node_unmanaged.go

```diff
@@ -374,88 +607,76 @@ func (tm *TestUnmanagedMiner) waitForMutableDeadline(ctx context.Context, sector
-		require.NoError(tm.t, err)
+		_, err := tm.FullNode.WaitTillChainOrError(tm.ctx, HeightAtLeast(targetEpoch+5))
+		req.NoError(err)
 	}
 }
 
-func (tm *TestUnmanagedMiner) NextSectorNumber() abi.SectorNumber {
-	sectorNumber := tm.currentSectorNum
-	tm.currentSectorNum++
-	return sectorNumber
-}
-
-func (tm *TestUnmanagedMiner) PrepareSectorForProveCommit(
-	ctx context.Context,
+func (tm *TestUnmanagedMiner) preCommitSectors(
+	sealRandEpoch abi.ChainEpoch,
 	proofType abi.RegisteredSealProof,
-	sectorNumber abi.SectorNumber,
-	pieces []abi.PieceInfo,
-) (seedEpoch abi.ChainEpoch, proveCommit []byte) {
-
-	req := require.New(tm.t)
-
-	// Wait for the pre-commitseal randomness to be available (we can only draw seal randomness from tipsets that have already achieved finality)
-	preCommitSealRandEpoch := tm.waitPreCommitSealRandomness(ctx, sectorNumber, proofType)
-
-	// Generate a Pre-Commit for the CC sector -> this persists the proof on the `TestUnmanagedMiner` Miner State
-	tm.generatePreCommit(ctx, sectorNumber, preCommitSealRandEpoch, proofType, pieces)
-
-	// --------------------Create pre-commit for the CC sector -> we'll just pre-commit `sector size` worth of 0s for this CC sector
+	sectors []sectorInfo,
+) error {
 
 	if !proofType.IsNonInteractive() {
-		// Submit the Pre-Commit to the network
-		var uc *cid.Cid
-		if len(pieces) > 0 {
-			unsealedCid := tm.unsealedCids[sectorNumber]
-			uc = &unsealedCid
-		}
-		r, err := tm.SubmitMessage(ctx, &miner14.PreCommitSectorBatchParams2{
-			Sectors: []miner14.SectorPreCommitInfo{{
+		// Submit the PreCommit to the network
+		spci := make([]miner14.SectorPreCommitInfo, len(sectors))
+		for i, sector := range sectors {
+			if sector.proofType != proofType {
+				return fmt.Errorf("sector proof type does not match PreCommit proof type")
+			}
+			sealedCid := sector.sealedCid
+			var uc *cid.Cid
+			if len(sector.pieces) > 0 {
+				unsealedCid := sector.unsealedCid
+				uc = &unsealedCid
+			}
+			spci[i] = miner14.SectorPreCommitInfo{
 				Expiration:    2880 * 300,
-				SectorNumber:  sectorNumber,
+				SectorNumber:  sector.SectorNumber,
 				SealProof:     proofType,
-				SealedCID:     tm.sealedCids[sectorNumber],
-				SealRandEpoch: preCommitSealRandEpoch,
+				SealedCID:     sealedCid,
+				SealRandEpoch: sealRandEpoch,
 				UnsealedCid:   uc,
-			}},
-		}, 1, builtin.MethodsMiner.PreCommitSectorBatch2)
-		req.NoError(err)
-		req.True(r.Receipt.ExitCode.IsSuccess())
-	}
-
-	// Generate a ProveCommit for the CC sector
-	var seedRandomness abi.InteractiveSealRandomness
-	seedEpoch, seedRandomness = tm.proveCommitWaitSeed(ctx, sectorNumber, proofType)
-
-	proveCommit = []byte{0xde, 0xad, 0xbe, 0xef} // mock prove commit
-	if !tm.mockProofs {
-		proveCommit = tm.generateProveCommit(ctx, sectorNumber, proofType, seedRandomness, pieces)
+			}
+			if len(sector.pieces) > 0 {
+				sectorCopy := sector
+				spci[i].UnsealedCid = &sectorCopy.unsealedCid
```

Comment:
trying to be loop-safe for the reference, I think I was originally avoiding a lint error but it seems redundant in the way this ended up.

I've removed this block since it's duplicated